### PR TITLE
Deploy ERC1155 contract and tokens microservice

### PR DESCRIPTION
**Pull request by awrichar**
Tokens Docker will need to be deployed before this can work, which implies a dependency on https://github.com/hyperledger-labs/firefly-tokens-erc1155/pull/3. **Edit:** this pre-req is now [done](https://github.com/hyperledger-labs/firefly-tokens-erc1155/pkgs/container/firefly-tokens-erc1155).

### CODE REVIEW

**Comment by awrichar** [comment]
Probably makes sense to confirm the config file syntax with https://github.com/hyperledger-labs/firefly/pull/154 before merging this (as there is an inter-dependency there) - but this can ultimately go in before that PR is merged (FireFly will simply ignore the new config).

File: internal/stacks/fireflyIdentity.go [hunk 1]
```diff
@@ -86,14 +88,16 @@ func (s *Stack) httpJSON(method, url string, body, result interface{}) (err erro
...
+
 	for _, member := range s.Members {
 		orgName := fmt.Sprintf("org_%s", member.ID)
 		nodeName := fmt.Sprintf("node_%s", member.ID)
 		ffURL := fmt.Sprintf("http://127.0.0.1:%d/api/v1", member.ExposedFireflyPort)
 		updateStatus(fmt.Sprintf("registering %s and %s", orgName, nodeName), spin)
 
 		registerOrgURL := fmt.Sprintf("%s/network/register/node/organization", ffURL)
-		err := s.httpJSONWithRetry(http.MethodPost, registerOrgURL, nil, nil)
+		err := s.httpJSONWithRetry(http.MethodPost, registerOrgURL, emptyObject, nil)
```
Comment (awrichar) [inline internal/stacks/fireflyIdentity.go:100]: I find it weird that this body is required, so I opened a bug on FireFly: https://github.com/hyperledger-labs/firefly/issues/155

**Review by nguyer — APPROVED** [review]
This looks good. I have refactored a bunch of this code in a different branch. That will be fun to sort through but I'll deal with that later.

**Comment by awrichar** [comment]
I moved some of the unrelated changes out to #81 and merged that. Will probably sit on the rest of this until @nguyer's refactor is complete, because it makes sense that tokens are abstracted in a similar manner to his blockchain work (and not hardcoded to always use ERC1155).

**Comment by awrichar** [comment]
Reworked this on top of the new changes from #82. Now includes logic for selecting a token provider (ERC1155 is the only valid option as of now).

**Comment by awrichar** [comment]
It occurs to me that maybe "no tokens" should also be a valid selection. Open to what that should look like, although I still lean toward including token support by default unless explicitly requested otherwise.

File: internal/stacks/stack_manager.go [hunk 2]
```diff
@@ -430,8 +439,8 @@ func checkPortAvailable(port int) (bool, error) {
 func (s *StackManager) runFirstTimeSetup(verbose bool, options *StartOptions) error {
 	workingDir := filepath.Join(constants.StacksDir, s.Stack.Name)
 
-	s.Log.Info("initializing blockchain node")
-	if err := s.blockchainProvider.RunFirstTimeSetup(); err != nil {
+	s.Log.Info("performing early initialization tasks")
+	if err := s.blockchainProvider.PreFirstTimeSetup(); err != nil {
```
Comment (nguyer) [inline internal/stacks/stack_manager.go:443]: I think both this and the tokens provider should have this function named the same thing. It feels strange to have a `PreFirstTimeSetup` without the `PostFirstTimeSetup` or vice versa.

**Review by nguyer — CHANGES_REQUESTED** [review]

File: internal/stacks/stack_manager.go (hunk 2 above)
Comment (awrichar) [reply internal/stacks/stack_manager.go:443]: Yea, I was drifting toward defining a common set of lifecycle hooks and letting all providers implement the ones they need - but that may be premature. I'll name them both `FirstTimeSetup` for now, even though they're invoked from different points. We can continue to revisit as the "providers" concept evolves.

**Review by nguyer — APPROVED** [review]

**Pull request merged**